### PR TITLE
TF-2859 add language to reply calendar event

### PR DESCRIPTION
**Pull request by tddang-linagora**
### Issue
- #2859 

### Demo

https://github.com/linagora/tmail-flutter/assets/160106668/57a5639d-79b6-433a-b837-e82304ce2d90

### CODE REVIEW

File: lib/features/email/presentation/controller/single_email_controller.dart [hunk 1]
```diff
@@ -1718,40 +1720,72 @@ class SingleEmailController extends BaseController with AppLoaderMixin {
...
+          mailboxDashBoardController.sessionCurrent!.capabilities[CapabilityIdentifier.jamesCalendarEvent] as CalendarEventCapability)));
     }
   }
 
   void _maybeCalendarEventAction() {
     if (_maybeCalendarEventInteractor == null
       || _displayingEventBlobId == null
-      || mailboxDashBoardController.accountId.value == null) {
+      || mailboxDashBoardController.accountId.value == null
+      || mailboxDashBoardController.sessionCurrent?.capabilities[CapabilityIdentifier.jamesCalendarEvent] == null
```
Comment (hoangdat) [inline lib/features/email/presentation/controller/single_email_controller.dart:1779]: how about use `getCapabilityProperties` method?

Comment (tddang-linagora) [reply lib/features/email/presentation/controller/single_email_controller.dart]: Done

File: lib/features/email/presentation/controller/single_email_controller.dart [hunk 2]
```diff
@@ -1718,40 +1720,78 @@ class SingleEmailController extends BaseController with AppLoaderMixin {
...
+
+    final currentLanguage = locale.languageCode;
+    if (supportedLanguages.contains(currentLanguage)) {
+      return currentLanguage;
+    } else if (supportedLanguages.contains('en')) {
+      return 'en';
+    } else {
+      return supportedLanguages.firstOrNull;
+    }
+  }
```
Comment (dab246) [inline lib/features/email/presentation/controller/single_email_controller.dart:1739]: IMO, Should be included in `session_extesion`

File: lib/features/email/presentation/controller/single_email_controller.dart [hunk 3]
```diff
@@ -1718,40 +1720,78 @@ class SingleEmailController extends BaseController with AppLoaderMixin {
...
   
   void _acceptCalendarEventAction() {
     if (_acceptCalendarEventInteractor == null
       || _displayingEventBlobId == null
-      || mailboxDashBoardController.accountId.value == null) {
-        consumeState(Stream.value(Left(CalendarEventAcceptFailure())));
+      || mailboxDashBoardController.accountId.value == null
+      || mailboxDashBoardController.sessionCurrent?.getCapabilityProperties<CalendarEventCapability>(
+        mailboxDashBoardController.accountId.value!,
+        CapabilityIdentifier.jamesCalendarEvent) == null
```
Comment (dab246) [inline lib/features/email/presentation/controller/single_email_controller.dart:1747]: IMO, should be create method in session_extension for validate it.

File: lib/features/email/presentation/controller/single_email_controller.dart [hunk 4]
```diff
@@ -1718,40 +1720,78 @@ class SingleEmailController extends BaseController with AppLoaderMixin {
...
 
   void _rejectCalendarEventAction() {
     if (_rejectCalendarEventInteractor == null
       || _displayingEventBlobId == null
-      || mailboxDashBoardController.accountId.value == null) {
-        consumeState(Stream.value(Left(CalendarEventRejectFailure())));
+      || mailboxDashBoardController.accountId.value == null
+      || mailboxDashBoardController.sessionCurrent?.getCapabilityProperties<CalendarEventCapability>(
+        mailboxDashBoardController.accountId.value!,
+        CapabilityIdentifier.jamesCalendarEvent) == null
```
Comment (dab246) [inline lib/features/email/presentation/controller/single_email_controller.dart:1766]: idem

File: lib/features/email/presentation/controller/single_email_controller.dart [hunk 5]
```diff
@@ -1718,40 +1720,78 @@ class SingleEmailController extends BaseController with AppLoaderMixin {
...
 
   void _maybeCalendarEventAction() {
     if (_maybeCalendarEventInteractor == null
       || _displayingEventBlobId == null
-      || mailboxDashBoardController.accountId.value == null) {
-        consumeState(Stream.value(Left(CalendarEventMaybeFailure())));
+      || mailboxDashBoardController.accountId.value == null
+      || mailboxDashBoardController.sessionCurrent?.getCapabilityProperties<CalendarEventCapability>(
+        mailboxDashBoardController.accountId.value!,
+        CapabilityIdentifier.jamesCalendarEvent) == null
```
Comment (dab246) [inline lib/features/email/presentation/controller/single_email_controller.dart:1785]: idem

File: contact/pubspec.lock [hunk 6]
```diff
@@ -531,7 +531,7 @@ packages:
     dependency: "direct main"
     description:
       path: "."
-      ref: "enhancement/calendar_event_capability_properties"
+      ref: master
       resolved-ref: "951ba341e68534fa054c1ef3df2338d074999712"
```
Comment (hoangdat) [inline contact/pubspec.lock:535]: IMO, it is not the right ref.

File: lib/features/email/presentation/controller/single_email_controller.dart (hunk 2 above)
Comment (tddang-linagora) [reply lib/features/email/presentation/controller/single_email_controller.dart]: Done

File: lib/features/email/presentation/controller/single_email_controller.dart (hunk 3 above)
Comment (tddang-linagora) [reply lib/features/email/presentation/controller/single_email_controller.dart]: Done

File: lib/features/email/presentation/controller/single_email_controller.dart (hunk 4 above)
Comment (tddang-linagora) [reply lib/features/email/presentation/controller/single_email_controller.dart]: Done

File: lib/features/email/presentation/controller/single_email_controller.dart (hunk 5 above)
Comment (tddang-linagora) [reply lib/features/email/presentation/controller/single_email_controller.dart]: Done

File: test/features/email/presentation/controller/single_email_controller_test.dart [hunk 7]
```diff
@@ -270,6 +278,14 @@ void main() {
   group('calendar event reply test:', () {
     final blobId = Id('abc123');
     final calendarEvent = CalendarEvent();
+    const locale = Locale('en', 'US');
+    final languageCacheManager = MockLanguageCacheManager();
+
+    setUp(() {
+      when(languageCacheManager.getStoredLanguage()).thenReturn(null);
```
[the remark this answers is not in the document]
Comment (tddang-linagora) [reply test/features/email/presentation/controller/single_email_controller_test.dart:285]: Test removed

File: contact/pubspec.lock (hunk 6 above)
Comment (tddang-linagora) [reply contact/pubspec.lock]: Fixed

File: model/lib/extensions/session_extension.dart [hunk 8]
```diff
@@ -63,4 +66,35 @@ extension SessionExtension on Session {
...
+  ({
+    bool isAvailable,
+    CalendarEventCapability? calendarEventCapability
+  }) validateCalendarEventCapability(AccountId accountId) {
+    final capability = getCapabilityProperties<CalendarEventCapability>(
+      accountId,
+      CapabilityIdentifier.jamesCalendarEvent);
+    
+    return (isAvailable: capability != null, calendarEventCapability: capability);
+  }
```
Comment (dab246) [inline model/lib/extensions/session_extension.dart:79]: somewhat confusing

Comment (tddang-linagora) [reply model/lib/extensions/session_extension.dart:79]: I use [Dart Record](https://dart.dev/language/records)

**Review by dab246 — APPROVED** [review]
lgtm

**Review by hoangdat — COMMENTED** [review]
lgtm, please rebase

File: test/features/email/presentation/controller/single_email_controller_test.dart (hunk 7 above)
[the remark this answers is not in the document]
Comment (hoangdat) [reply test/features/email/presentation/controller/single_email_controller_test.dart:294]: still here?, please remove

[the remark this answers is not in the document]
Comment (tddang-linagora) [reply test/features/email/presentation/controller/single_email_controller_test.dart]: Done

**Review by hoangdat — APPROVED** [review]

**Pull request merged**